src/averageAges.js

```diff
@@ -15,11 +15,18 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const arrayMen = people.filter(el => el.sex === 'm');
+
+  if (!century) {
+    return +(arrayMen.map((el) => el.died - el.born)
+      .reduce((acc, el) => acc + el) / arrayMen.length).toFixed(2);
+  }
+
+  const arrayMenByCentiry = arrayMen
+    .filter(el => Math.ceil(el.died / 100) === century);
+
+  return +(arrayMenByCentiry.map(el => el.died - el.born)
+    .reduce((acc, el) => acc + el) / arrayMenByCentiry.length).toFixed(2);
```

Comment:
you do it several time, you can create some util function, that will accept array, and do something with it